runtime/src/iree/hal/buffer_view.h

```diff
@@ -48,6 +48,13 @@ enum iree_hal_numerical_type_bits_t {
   IREE_HAL_NUMERICAL_TYPE_FLOAT_BRAIN = IREE_HAL_NUMERICAL_TYPE_FLOAT | 0x02u,
   // Paired (real, imag) complex number in floating-point format.
   IREE_HAL_NUMERICAL_TYPE_FLOAT_COMPLEX = IREE_HAL_NUMERICAL_TYPE_FLOAT | 0x03u,
+  // Ad-hoc entries for the zoo of low-bit-depth float types. There currently
+  // are 4 8-bit float types that we need to support. For now just create these
+  // 4 nameless "slots" to fit those types.
+  IREE_HAL_NUMERICAL_TYPE_FLOATISH_0 = IREE_HAL_NUMERICAL_TYPE_FLOAT | 0x04u,
```

Comment:
"FLOATISH" is cute but we want something more precise here. If these are pinned to specific encodings then just name them the specific encoding (no need for a mapping)